akka23-conductr-client-lib/src/main/scala/com/typesafe/conductr/clientlib/akka/BundlesConnector.scala

```diff
@@ -6,31 +6,108 @@
+   *
+   * @param conductrAddress the ConductR Control Protocol base URL
+   * @param apiVersion the API version of the Control Protocol
+   * @param connectUntil if returning `true` before the timeout specified by `stopAfter` has elapsed, the resulting
+   *                     `Future[Unit]` will be completed.
+   * @param stopAfter the timeout waiting for the `connectUntil` condition to become `true`.
+   * @param expectingEventBurstTimeout the expected amount of time of which SSE event burst from `/bundles/events` will
+   *                                   take place. Once event burst is over, the latest bundle state will be obtained
+   *                                   from `/bundles/events`
+   * @param reconnectInterval the amount of time to wait to reconnect to `/bundles/events` when disconnected.
+   * @param system the actor system which is used to build [[BundlesConnector]]
+   * @return the tuple of [[ActorRef]] to the [[BundlesConnector]] and `Future[Unit]`.
+   *         The `Future[Unit]` will be completed when `connectUntil` returns `true` before timeout specified by
+   *         `stopAfter` has elapsed.
+   */
+  def connect(
+    conductrAddress: URL,
+    apiVersion: ApiVersion.Value,
+    connectUntil: Seq[Bundle] => Boolean,
+    stopAfter: FiniteDuration,
+    expectingEventBurstTimeout: FiniteDuration = 500.millis,
+    reconnectInterval: FiniteDuration = 500.millis
+  )(implicit system: ActorRefFactory): (ActorRef, Future[Unit]) = {
+    val promise = Promise[Unit]
+    val connector = system.actorOf(
+      props(
+        conductrAddress,
+        apiVersion,
+        connectUntil,
+        onComplete = () => promise.success(Unit),
+        stopAfter = Some(stopAfter),
+        onStop = Some(() => promise.failure(TimeoutException))
+      )
+    )
+    connector -> promise.future
+  }
+
+  /**
+   * Client API which provides a nicer interface to the [[BundlesConnector]] actor.
+   *
+   * Provides a means for the caller to connect to the ConductR Bundles endpoint continuously.
+   *
+   * Returns an [[ActorRef]] to the [[BundlesConnector]], which can be used by the caller to terminate the connection
+   * to the ConductR Bundles endpoint by stopping the actor.
+   *
+   * @param conductrAddress the ConductR Control Protocol base URL
+   * @param apiVersion the API version of the Control Protocol
+   * @param expectingEventBurstTimeout the expected amount of time of which SSE event burst from `/bundles/events` will
+   *                                   take place. Once event burst is over, the latest bundle state will be obtained
+   *                                   from `/bundles/events`
+   * @param reconnectInterval the amount of time to wait to reconnect to `/bundles/events` when disconnected.
+   * @param system the actor system which is used to build [[BundlesConnector]]
+   * @return [[ActorRef]] to the bundles connector.
    */
-  type Callback = Seq[Bundle] => CallbackResult
+  def continuously(
```

Comment:
So we don't presently use this - you're thinking of the conductr-haproxy use-case right?